### PR TITLE
Implement basic version of getDatabase in catalog service

### PR DESCRIPTION
**Pull request by yuzhu**
Note here `getDatabase` returns an object that contains only the database name.
A subsequent PR will add more content into this database object and the associated Udb object to populate this database object.

### CODE REVIEW

**Comment by yuzhu** [comment]
@ZacBlanco PTAL. I implemented bare minimum here, but we probably want to have a more complete database object and Udb object with information about the database owner and params. That's for another PR.

File: table/server/master/src/main/java/alluxio/master/table/AlluxioCatalog.java [hunk 1]
```diff
@@ -196,6 +196,19 @@ private Table getTableInternal(String dbName, String tableName) throws IOExcepti
     return new ArrayList<>(mDBs.keySet());
   }
 
+  /**
+   * Get Database by its name.
+   *
+   * @param dbName database name
+   * @return a databae object
```
Comment (ZacBlanco) [inline table/server/master/src/main/java/alluxio/master/table/AlluxioCatalog.java]: nit: typo

**Comment by ZacBlanco** [comment]
Does our current database object only contain the name? I thought I saw some fields for description and location before

**Comment by yuzhu** [comment]
> Does our current database object only contain the name? I thought I saw some fields for description and location before

The RPC proto contains description and location, but we don't store that information in our server side representation or our Udb object from Hive yet. Hopefully this is enough to get past the test for now. Changing Udb requires changes to journal as well,  so that's why I want to make that a separate PR.

**Comment by ZacBlanco** [comment]
Also can we add some kind of simple UT for the the `getDatabase` call?

File: tests/src/test/java/alluxio/server/ft/journal/TableMasterJournalIntegrationTest.java [hunk 2]
```diff
@@ -92,8 +96,12 @@ public void journalAttachDb() throws Exception {
     TableMaster tableMaster =
         mCluster.getLocalAlluxioMaster().getMasterProcess().getMaster(TableMaster.class);
 
+    mException.expect(IOException.class);
+    tableMaster.getDatabase(DB_NAME);
```
Comment (ZacBlanco) [inline tests/src/test/java/alluxio/server/ft/journal/TableMasterJournalIntegrationTest.java]: won't this stop the rest of the test from being run?

File: table/server/master/src/test/java/alluxio/master/table/AlluxioCatalogTest.java [hunk 3]
```diff
@@ -100,6 +100,18 @@ public void detachDb() throws Exception {
     assertEquals(0, mCatalog.getAllDatabases().size());
   }
 
+  @Test
+  public void getDb() throws Exception {
+    String dbName = "testdb";
+    TestDatabase.genTable(1, 2);
+    mException.expect(NotFoundException.class);
+    mCatalog.getDatabase(dbName);
```
Comment (ZacBlanco) [inline table/server/master/src/test/java/alluxio/master/table/AlluxioCatalogTest.java]: similarly, won't all lines after this not be executed?

File: tests/src/test/java/alluxio/server/ft/journal/TableMasterJournalIntegrationTest.java (hunk 2 above)
Comment (yuzhu) [reply tests/src/test/java/alluxio/server/ft/journal/TableMasterJournalIntegrationTest.java]: doh.. why aren't we on Junit 5 yet?  I totally misunderstood the semantics of `expect`

File: table/server/master/src/main/java/alluxio/master/table/TableMaster.java [hunk 4]
```diff
@@ -138,4 +139,12 @@ boolean detachDatabase(String dbName)
...
    */
   boolean syncDatabase(String dbName) throws IOException;
+
+  /**
+   * Gets a database object.
+   *
+   * @param dbName the database name
+   * @return a database object
+   */
+  Database getDatabase(String dbName) throws IOException;
```
Comment (ZacBlanco) [inline table/server/master/src/main/java/alluxio/master/table/TableMaster.java]: i don't care that much but for the most part methods are in alphabetical order in this class. Should we try to keep it that way? the transform methods are obviously not in the right order, but the rest seem to be

**Comment by yuzhu** [comment]
alluxio-bot, merge this please

**Comment by yuzhu** [comment]
alluxio-bot, cherry-pick this to branch-2.1 please

**Pull request merged**